now.go

```diff
@@ -1,9 +1,30 @@
 package utc
 
-import "time"
+import (
+	"sync"
+	"time"
+)
 
-// nowFn is the function used to get the current time and can be mocked with MockNow/MockNowFn
-var nowFn = now
+// nowFnMu protects access to nowFn
+var nowFnMu sync.Mutex
+
+func init() {
+	nowFn = now
+	setNowFn(now)
+}
+
+// allowClock allows using a custom Clock and is called during test whenever setClock is called.
+func allowClock() {
```

Comment:
As far as I can tell, this is always called because it gets called on `init` by `setNowFn`. In that case, we can take out this complexity and move it to the `race` file.

Reply:
Good catch, thanks: actually `init` does not need to call `setNowFn`.